Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -0,0 +1,198 @@
+      "inputs": {
+        "Service": "ec2",
+        "Api": "DescribeInstances",
+        "InstanceIds": [
+          "{{instanceId}}"
+        ]
+      },
+      "outputs": [
+        {
+          "Name": "PlacementAvailabilityZone",
+          "Selector": "$.Reservations[0].Instances[0].Placement.AvailabilityZone",
+          "Type": "String"
+        }
+      ],
+      "name": "extractAvailabilityZone",
+      "action": "aws:executeAwsApi",
+      "timeoutSeconds": 30,
+      "onFailure": "Abort"
+    },
+    {
+      "maxAttempts": 3,
+      "inputs": {
+        "DocumentName": "AWS-CreateSnapshot",
+        "RuntimeParameters": {
+          "VolumeId": "{{extractEBSvolumeID.EBSVolumeID}}"
+        }
+      },
+      "name": "CreateSnapshot",
+      "action": "aws:executeAutomation",
+      "timeoutSeconds": 3600,
+      "onFailure": "Abort"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "Service": "ec2",
+        "Api": "DescribeSnapshots",
+        "SnapshotIds": "{{CreateSnapshot.Output}}"
+      },
+      "outputs": [
+        {
+          "Name": "SNAPSHOTID",
+          "Selector": "$.Snapshots[0].SnapshotId",
+          "Type": "String"
+        }
+      ],
+      "name": "extractSnapshotID",
+      "action": "aws:executeAwsApi",
+      "timeoutSeconds": 30,
+      "onFailure": "Abort"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "Service": "ec2",
+        "Api": "CopySnapshot",
+        "SourceSnapshotId": "{{extractSnapshotID.SNAPSHOTID}}",
+        "SourceRegion": "{{region}}",
+        "Encrypted": true,
+        "KmsKeyId": "{{KmsKeyId}}",
+        "DestinationRegion": "{{region}}"
+      },
+      "outputs": [
+        {
+          "Name": "EncryptedSnapshotID",
+          "Selector": "$.SnapshotId",
+          "Type": "String"
+        }
+      ],
+      "name": "CopySnapshot",
+      "action": "aws:executeAwsApi",
+      "timeoutSeconds": 3600,
+      "onFailure": "Abort"
+    },
+    {
+      "inputs": {
+        "Duration": "PT2M"
+      },
+      "name": "sleep1",
+      "action": "aws:sleep"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "Service": "ec2",
+        "Api": "CreateVolume",
+        "AvailabilityZone": "{{extractAvailabilityZone.PlacementAvailabilityZone}}",
+        "Encrypted": true,
+        "KmsKeyId": "{{KmsKeyId}}",
+        "SnapshotId": "{{CopySnapshot.EncryptedSnapshotID}}",
+        "VolumeType": "gp2"
+      },
+      "outputs": [
+        {
+          "Name": "NewRootVolumeID",
+          "Selector": "$.VolumeId",
+          "Type": "String"
+        }
+      ],
+      "name": "CreateVolume",
+      "action": "aws:executeAwsApi",
+      "timeoutSeconds": 30,
+      "onFailure": "Abort"
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "DocumentName": "AWS-StopEC2Instance",
+        "RuntimeParameters": {
+          "InstanceId": "{{instanceId}}"
+        }
+      },
+      "name": "StopInstance",
```

Comment:
Step names are usually camelCase in SSM.

Reply:
Done